airbyte-integrations/connectors/source-railz-ai/source_railz_ai/components.py

```diff
@@ -0,0 +1,370 @@
+            self._cursor[stream_slice["businessName"]][stream_slice["serviceName"]] = {}
+        service = self._cursor[stream_slice["businessName"]][stream_slice["serviceName"]]
+
+        last_record_cursor_value = last_record[self._datetime_cursor_field]
+
+        if self._datetime_cursor_field not in service or service[self._datetime_cursor_field] < last_record_cursor_value:
+            service[self._datetime_cursor_field] = last_record_cursor_value
+
+    def stream_slices(self, sync_mode: SyncMode, stream_state: StreamState) -> Iterable[StreamSlice]:
+        for stream_slice in super().stream_slices(sync_mode, stream_state):
+            state_cursor_value = stream_state \
+                .get(stream_slice["businessName"], {}).get(stream_slice["serviceName"], {}).get(self._datetime_cursor_field, None)
+            if state_cursor_value:
+                start = self._datetime_parse_date(state_cursor_value) \
+                    .strftime(self.base_datetime_stream_slicer_options["start_datetime"]["datetime_format"])
+                datetime_stream_slicer = self._create_datetime_stream_slicer(
+                    {
+                        **self.base_datetime_stream_slicer_options,
+                        "start_datetime": {
+                            **self.base_datetime_stream_slicer_options["start_datetime"],
+                            "datetime": start,
+                        },
+                    }
+                )
+            else:
+                datetime_stream_slicer = self._create_datetime_stream_slicer(self.base_datetime_stream_slicer_options)
+            for datetime_stream_slice in datetime_stream_slicer.stream_slices(
+                sync_mode, stream_state.get(stream_slice["businessName"], {}).get(stream_slice["serviceName"], {})
+            ):
+                yield {**stream_slice, **datetime_stream_slice}
+
+    def get_request_params(
+        self,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        options = {}
+        if stream_slice:
+            start_date = (stream_slice[self._datetime_slice_field_start],)
+            end_date = (stream_slice[self._datetime_slice_field_end],)
+            if isinstance(start_date, tuple):
+                start_date = start_date[0]
+            if isinstance(end_date, tuple):
+                end_date = end_date[0]
+
+            options = {
+                "businessName": stream_slice["businessName"],
+                "serviceName": stream_slice["serviceName"],
+                self._datetime_cursor_field: f"gte:{start_date};lte:{end_date}",
+            }
+        return options
+
+    @property
+    def _datetime_cursor_field(self):
+        datetime_stream_slicer = self._create_datetime_stream_slicer(self.base_datetime_stream_slicer_options)
```

Comment:
why are we creating a new stream slicer at every call instead of just one in the init?